packages/server/src/instances/clearing/service.ts

```diff
@@ -53,7 +54,22 @@ export class InstanceClearingService extends Service {
     this.destroyed = true
   }
 
-  private async handleCacheDispose(key: string) {
+  private handleCacheDispose(key: string, value: any) {
+    if (this.destroyed) {
+      return
+    }
+
+    // if it's been deleted intentionally we assume disposing was done
+    if (this.statesDeleted) {
+      delete this.statesDeleted[key]
+      return
+    }
```

Comment:
I added this here because the dispose method from the lru gets called from values that are removed intentionnaly as well. This was causing two health events of type sleep to be emitted, so I put this safety here.